landlab/components/landslides/landslide_probability.py

```diff
@@ -485,39 +583,77 @@ def calculate_factor_of_safety(self, i):
-                np.log(
-                    (self._recharge_stdev[i] ** 2) / (self._recharge_mean[i] ** 2) + 1
+        # Create a switch to imply whether Recharge or Depth to Water Table Forcing
+        if self._groundwater__depth_distribution is not None:
+
+            # Depth to water table distribution based on distribution type
+            if self._groundwater__depth_distribution == "data_driven_spatial":
+
+                self._calculate_HSD_groundwater_depth(i)
+
+            elif self._groundwater__depth_distribution == "lognormal_spatial":
+                mu_lognormal = np.log(
+                    (self._depth_mean[i] ** 2)
+                    / np.sqrt(self._depth_stdev[i] ** 2 + self._depth_mean[i] ** 2)
                 )
-            )
-            self._Re = np.random.lognormal(mu_lognormal, sigma_lognormal, self._n)
-            self._Re /= 1000.0  # Convert mm to m
+                sigma_lognormal = np.sqrt(
+                    np.log((self._depth_stdev[i] ** 2) / (self._depth_mean[i] ** 2) + 1)
+                )
+                self._De = np.random.lognormal(mu_lognormal, sigma_lognormal, self._n)
+
+            self._hw_dist = self._hs_mode - self._De
+            self._hw_dist[
+                np.where(self._hw_dist < 0)
+            ] = 0  # no water in soil column when De input > soil thickness
+
+            # output mean depth of water
+            self._soil__mean_watertable_depth = np.mean(self._De)
+
+        if self._groundwater__recharge_distribution is not None:
+            # recharge distribution based on distribution type
+            if self._groundwater__recharge_distribution == "data_driven_spatial":
+                self._calculate_HSD_recharge(i)
+                self._Re /= 1000.0  # mm->m
+
+            elif self._groundwater__recharge_distribution == "lognormal_spatial":
+                mu_lognormal = np.log(
+                    (self._recharge_mean[i] ** 2)
+                    / np.sqrt(
+                        self._recharge_stdev[i] ** 2 + self._recharge_mean[i] ** 2
+                    )
+                )
+                sigma_lognormal = np.sqrt(
+                    np.log(
+                        (self._recharge_stdev[i] ** 2) / (self._recharge_mean[i] ** 2)
+                        + 1
+                    )
+                )
+                self._Re = np.random.lognormal(mu_lognormal, sigma_lognormal, self._n)
+                self._Re /= 1000.0  # Convert mm to m
+
+            # output mean recharge
+            self._soil__mean_recharge = np.mean(self._Re)
 
         # Cohesion
-        # if don't provide fields of min and max C, uncomment 2 lines below
-        #    Cmin = self._Cmode-0.3*self._Cmode
-        #    Cmax = self._Cmode+0.3*self._Cmode
-        self._C = np.random.triangular(
-            self._Cmin, self._Cmode, self._Cmax, size=self._n
-        )
+        if np.all(self._grid.at_node["soil__minimum_total_cohesion"]) is not None:
```

Comment:
I suspect that this is always going to be the case. np.all will return True or False and the field is a required input (e.g. init will fail if it is not present).

So I don't think the code will ever go into the else block. Once the tests pass, I'd recommend looking carefully at the coverage and removing code that is never used (e.g., the else block below).